File: task-03/python/Sub-Task3/code3.py
def print_diamond(n):
    # Upper half of the diamond
    for i in range(1, n + 1):
        spaces = ' ' * (n - i)
        stars = '*' * (2 * i - 1)
        print(spaces + stars)

    # Lower half of the diamond
    for i in range(n - 1, 0, -1):
        spaces = ' ' * (n - i)
        stars = '*' * (2 * i - 1)
        print(spaces + stars)

def main():
    # Prompt user for input
    n = int(input("Enter the number of rows for the diamond pattern: "))

    # Validate input
    if n <= 0:
        print("The number of rows must be greater than 0.")
    else:
        # Print the diamond pattern
        print_diamond(n)

if __name__ == "__main__":
    main()

File: task-03/Haskell/Sub-Task3/code3.hs
import Control.Monad (when)

-- Function to print a single line of the diamond
printLine :: Int -> Int -> IO ()
printLine n i = do
    let spaces = replicate (n - i) ' '
    let stars = replicate (2 * i - 1) '*'
    putStrLn (spaces ++ stars)

-- Function to print the diamond pattern
printDiamond :: Int -> IO ()
printDiamond n = do
    -- Upper half of the diamond
    mapM_ (printLine n) [1..n]
    -- Lower half of the diamond
    mapM_ (printLine n) [n-1,n-2..1]

main :: IO ()
main = do
    -- Prompt user for input
    putStr "Enter the number of rows for the diamond pattern: "
    input <- getLine
    let n = read input :: Int

    -- Validate input
    when (n <= 0) $ do
        putStrLn "The number of rows must be greater than 0."
        return ()

    -- Print the diamond pattern
    printDiamond n

